test/core/utils/respondable.js

```diff
@@ -1,515 +1,561 @@
         done();
-      }
-    };
-
-    axe.utils.respondable(win, 'batman', 'nananana');
+      }, done)
+    );
+    respondable(frameWin, 'greeting', 'hello', 'truthy');
   });
 
-  it('should add the `keepalive`', function(done) {
-    var win = {
-      postMessage: function(msg) {
-        msg = JSON.parse(msg);
-
-        assert.equal(msg._keepalive, 'batman');
+  it('passes a falsy keepalive value', function(done) {
+    frameSubscribe(
+      'greeting',
+      captureError(function(_, keepalive) {
+        assert.isFalse(keepalive);
         done();
-      }
-    };
+      }, done)
+    );
+    respondable(frameWin, 'greeting', 'hello', 0);
+  });
 
-    axe.utils.respondable(win, 'superman', 'spidey', 'batman');
+  it('can not publish to a parent frame', function(done) {
+    var isCalled = false;
+    axe.utils.respondable.subscribe('greeting', function() {
+      isCalled = true;
+    });
+    assert.throws(function() {
+      frameWin.axe.utils.respondable(window, 'greeting', 'hello', 0);
+    });
+    setTimeout(
+      captureError(function() {
+        assert.isFalse(isCalled);
+        done();
+      }, done),
+      100
+    );
   });
 
-  it('should add `_respondable` to the message', function(done) {
-    var win = {
-      postMessage: function(msg) {
-        msg = JSON.parse(msg);
+  it('does not expose private methods', function() {
+    var methods = Object.keys(respondable).sort();
+    assert.deepEqual(methods, ['subscribe', 'isInFrame'].sort());
+  });
 
-        assert.equal(msg._respondable, true);
+  it('passes serialized information only', function(done) {
+    var div = document.createElement('div');
+    frameSubscribe(
+      'greeting',
+      captureError(function(message) {
+        assert.deepEqual(message, {});
         done();
-      }
-    };
+      }, done)
+    );
 
-    axe.utils.respondable(win, 'batman', 'nananana');
+    respondable(frameWin, 'greeting', div);
   });
 
-  describe('messageHandler', function() {
-    var event = document.createEvent('Event');
-    // Define that the event name is 'build'.
-    event.initEvent('message', true, true);
-    event.source = window;
-
-    var eventData;
-    var win;
-    var axeVersion;
-
-    beforeEach(function() {
-      axeVersion = axe.version;
-      win = {
-        postMessage: function(message) {
-          var data = JSON.parse(message);
-          eventData.uuid = data.uuid;
-          event.data = JSON.stringify(eventData);
-          document.dispatchEvent(event);
-        }
-      };
-    });
-
-    afterEach(function() {
-      axe.version = axeVersion;
+  describe('subscribe', function() {
+    it('is called with the same topic', function(done) {
+      var doneOnce = once(done);
+      frameSubscribe('greeting', function() {
+        doneOnce();
+      });
+      respondable(frameWin, 'greeting');
+      afterMessage(frameWin, function() {
+        doneOnce(new Error('subscribe was not called'));
+      });
```

Comment:
This test is a bit confusing because it doesn't assert anything. Would be a bit more understandable if you used sinon to assert called.

```suggestion
      var spy = sinon.spy();
      frameSubscribe('greeting', spy);
      respondable(frameWin, 'greeting');
      afterMessage(frameWin, function() {
        captureError(function() {
          assert.isTrue(spy.called);
          done();
        }, done)
      });
```